webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/servers/RemoveNodeService.scala

```diff
@@ -78,26 +83,73 @@ import com.normation.rudder.repository.ldap.LDAPEntityMapper
 import com.normation.rudder.repository.ldap.ScalaReadWriteLock
 import com.normation.rudder.services.nodes.NodeInfoService
 import com.normation.rudder.services.nodes.NodeInfoService.A_MOD_TIMESTAMP
+import com.normation.rudder.services.nodes.NodeInfoServiceCached
 import com.normation.rudder.services.policies.write.NodePoliciesPaths
 import com.normation.rudder.services.policies.write.PathComputer
+import com.normation.rudder.services.servers.DeletionResult._
 import com.unboundid.ldap.sdk.Modification
 import com.unboundid.ldap.sdk.ModificationType
 import com.unboundid.ldif.LDIFChangeRecord
 import net.liftweb.common.Box
-import net.liftweb.common.EmptyBox
-import net.liftweb.common.Failure
-import net.liftweb.common.Full
 import zio._
+import zio.syntax._
+import zio.stream._
+import com.normation.zio._
 import org.joda.time.DateTime
 
 sealed trait DeletionResult
 object DeletionResult {
   final case class  PreHookFailed ( hookError : HookReturnCode.Error) extends DeletionResult
   final case class  PostHookFailed( hookError : HookReturnCode.Error) extends DeletionResult
-  final case object Success extends DeletionResult
+  final case object Success                                           extends DeletionResult
+  final case class  Error(err: RudderError)                           extends DeletionResult
+
+  def merge(one: DeletionResult, other: List[DeletionResult]): IOResult[Unit] = {
+    if((one :: other).forall(_ == Success)) UIO.unit
+    else {
+      val errors = (one::other).filterNot(_ == Success).map(e => e match {
+        case Error(err)          => err
+        case PreHookFailed(err)  => Inconsistency(s"Pre hook error: ${err.msg}")
+        case PostHookFailed(err) => Inconsistency(s"Post hook error: ${err.msg}")
+        case Success             => Unexpected(s"Success in deletionResult error processing: this case should node be reached, please report a bug.")
+      })
+      Accumulated(NonEmptyList.fromListUnsafe(errors)).fail // we know it's non-empty since they are not success
+    }
+  }
+}
+
+sealed trait DeleteMode { def name: String }
+final object DeleteMode {
+
+  final case object MoveToRemoved extends DeleteMode { val name = "move"}
```

Comment:
typo: MoveToRemove

Reply:
No it's ok, the tree is "Removed Inventory"